templates/components/form/form.jsx

```diff
@@ -67,12 +79,23 @@ var Form = React.createClass({
      var fields = Object.keys(this.props.fields).map(this.buildFormElement);
      return <div role="form">{fields}</div>;
   },
-  getClasses: function(field) {
+  getInputClasses: function(field) {
+    var isValid = this.isValid(field);
     return React.addons.classSet({
       'form-control': true,
-      'has-error': !this.isValid(field),
+      'has-error': !isValid,
+      'is-valid': isValid
     });
   },
+  getLabelClasses: function(field) {
+    var classes = {};
+    var isValid = this.isValid(field);
+    var ref = this.refs[field + 'Input'];
+    classes[this.getIconClass(field)] = true;
+    classes[this.getIconClass('error')] = !isValid;
```

Comment:
`this.getIconClass('error')` and `this.getIconClass('valid')` are invariant. There's not much point in constantly looking them up, we should just compute these values during componentWillMount or something and then refer to those